src/ringbuffer.rs

```diff
@@ -70,16 +71,15 @@ impl<T: Slot + Send> RingBuffer<T> {
     // Unsafe because we have no guarantees the caller won't invalidate this slot
     pub unsafe fn get(&self, from: usize, size: usize) -> &[T] {
         debug!("              RingBuffer get({}, {})", from, size);
-        let v : *mut Vec<T> = self.entries.get();
-        ::std::slice::from_raw_parts(v.offset(from), size)
+        &self.entries.get().as_ref().unwrap()[from .. from + size]
     }
 
     // Unsafe because we have no guarantees the caller won't invalidate this slot
     pub unsafe fn write(&self, position: usize, data: T) {
-        let v: *mut Vec<T> = self.entries.get();
-        let slot = v.get_mut(position);
-        drop(slot);
-        slot = data;
+        let v = self.entries.get();
+        let mut slot = v.as_mut().unwrap().get_mut(position).unwrap();
+        //TODO: Do we need to `drop` the current data?
```

Comment:
I think a drop is needed, but I could be wrong.  In the current setup, all event processors (even the final, "root" EP) receive borrowed references to data owned by the ring buffer.  So it's up to the ring to call the destructor for any owned data.  If we just replace the slot, I think we'll leak memory?